test/frontdesk_test.rb

```diff
@@ -0,0 +1,161 @@
+require_relative 'test_helper'
+
+describe "FrontDesk class" do
+  before do
+    @range = Hotel::DateRange.new(start_date: "1st Apr 2020", end_date: "3rd Apr 2020")
+    @front_desk = Hotel::FrontDesk.new
+  end
+
+  describe "#initialize" do
+    it "creates an instance of FrontDesk class" do
+      expect(@front_desk).must_be_instance_of Hotel::FrontDesk
+    end
+
+    it "has 20 items in a rooms list" do
+      expect(@front_desk.rooms.length).must_equal 20
+    end
+
+    it "creates a list of Room instances" do
+      expect(@front_desk.rooms.last).must_be_instance_of Hotel::Room
+    end
+
+    it "gives a correct room number for the first instance in the rooms list" do
+      expect(@front_desk.rooms.first.number).must_equal 1
+    end
+
+    it "gives a correct room number for the last instance in the rooms list" do
+      expect(@front_desk.rooms.last.number).must_equal 20
+    end
+
+    it "creates a list of reservations" do
+      expect(@front_desk.reservations).must_be_instance_of Array
+    end
+  end
+
+  describe "#get_bookings" do
+    before do
+      @front_desk.reserve_room(@range)
+    end
+
+    it "gives an instance of Reservation in a list of reservations" do
+      expect(@front_desk.get_bookings("2 Apr 2020").first).must_be_instance_of Hotel::Reservation
+    end
+
+    it "gives nil if a specified date coinsides with a check-out date" do
+      expect(@front_desk.get_bookings("5 Apr 2020").first).must_be_nil
+    end
+
+    it "returns an empty array if no reservations are found" do
+      expect(@front_desk.get_bookings("5 Apr 2020")).must_equal []
+    end
+  end
+
+  describe "#get_room_bookings" do
+    it "returns an instance of Reservation in a list" do
+      @front_desk.reserve_room(@range)
+      expect(@front_desk.get_room_bookings(1, @range).first).must_be_instance_of Hotel::Reservation
+    end
+  end
+
+  describe "#get_available_rooms" do
```

Comment:
Make sure to include a test for the situation where there are no rooms available. You test for this in your `reserve_room` tests, but a direct test of `available_rooms` in this situation is also important.